openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BasicAuthenticator.java

```diff
@@ -0,0 +1,477 @@
+              uriInfo.getRequestUri().getHost(),
+              user.getFullyQualifiedName(),
+              mailVerificationToken);
+      try {
+        EmailUtil.getInstance().sendEmailVerification(emailVerificationLink, user);
+      } catch (Exception e) {
+        LOG.error("Error in sending mail to the User : {}", e.getMessage());
+        throw new CustomExceptionMessage(424, EMAIL_SENDING_ISSUE);
+      }
+      // insert the token
+      tokenRepository.insertToken(emailVerificationToken);
+    }
+  }
+
+  @Override
+  public void sendPasswordResetLink(UriInfo uriInfo, User user, String subject, String templateFilePath)
+      throws IOException {
+    UUID mailVerificationToken = UUID.randomUUID();
+    PasswordResetToken resetToken = TokenUtil.getPasswordResetToken(user.getId(), mailVerificationToken);
+    LOG.info("Generated Password Reset verification token [" + mailVerificationToken + "]");
+    String passwordResetLink =
+        String.format(
+            "%s://%s/users/password/reset?user=%s&token=%s",
+            uriInfo.getRequestUri().getScheme(),
+            uriInfo.getRequestUri().getHost(),
+            user.getFullyQualifiedName(),
+            mailVerificationToken);
+    try {
+      EmailUtil.getInstance().sendPasswordResetLink(passwordResetLink, user, subject, templateFilePath);
+    } catch (Exception e) {
+      LOG.error("Error in sending mail to the User : {}", e.getMessage());
+      throw new CustomExceptionMessage(424, EMAIL_SENDING_ISSUE);
+    }
+    // don't persist tokens delete existing
+    tokenRepository.deleteTokenByUserAndType(user.getId().toString(), PASSWORD_RESET.toString());
+    tokenRepository.insertToken(resetToken);
+  }
+
+  @Override
+  public void resetUserPasswordWithToken(UriInfo uriInfo, PasswordResetRequest request) throws IOException {
+    String tokenID = request.getToken();
+    PasswordResetToken passwordResetToken = (PasswordResetToken) tokenRepository.findByToken(tokenID);
+    if (passwordResetToken == null) {
+      throw new EntityNotFoundException("Invalid Password Request. Please issue a new request.");
+    }
+    List<String> fields = userRepository.getAllowedFieldsCopy();
+    fields.add(USER_PROTECTED_FIELDS);
+    User storedUser =
+        userRepository.getByName(
+            uriInfo, request.getUsername(), new EntityUtil.Fields(fields, String.join(",", fields)));
+    // token validity
+    if (!passwordResetToken.getUserId().equals(storedUser.getId())) {
+      throw new RuntimeException("Token does not belong to the user.");
+    }
+    verifyPasswordResetTokenExpiry(passwordResetToken);
+    // passwords validity
+    if (!request.getPassword().equals(request.getConfirmPassword())) {
+      throw new RuntimeException("Password and Confirm Password should match");
+    }
+    PasswordUtil.validatePassword(request.getPassword());
+
+    String newHashedPwd = BCrypt.withDefaults().hashToString(12, request.getPassword().toCharArray());
```

Comment:
what is the 12 number is about here? can we add some comments and add a final constant